### ISSUE TITLE
Feature request: adding option for additional classes

### ISSUE THREAD

**Issue by ethanbass** [SchornacklabSLCU/amfinder#20]
I am wondering what would be involved in adding support for additional classes of tiles. Specifically, I would be interested in being able to train a CNN to distinguish colonization by AMF from colonization by other root endophytes. A lot of our roots from the field are colonized by "dark septate endophytes" which appear quite different from AMF. For example, I believe the root on the left in the attached picture contains vesicles from AMF, but the root on the right contains microsclerotia from DSE. It would be great to be able to train a CNN to differentiate these different types of fungi
<img width="803" alt="Screen Shot 2022-02-17 at 1 04 03 PM" src="https://user-images.githubusercontent.com/15876283/154543589-8857e77f-fcc6-45eb-bb98-21b5ccbe6aa7.png">
!

**Comment by EEvangelisti** [comment]
Dear Ethan,

Support for an arbitrary number of classes would indeed be a cool feature. It should be easy to implement in `amf`. However, updating the user interface (`amfbrowser`) would be more challenging. Therefore, I would only consider implementing such a feature as part of a collaboration.

There is an easy workaround for the specific purpose of identifying one more category (microsclerotia). You may have noticed from the AMFinder publication that the training dataset did not contain enough hyphopodia to efficiently predict this class of hyphal structures. You can therefore repurpose it to label microsclerotia. Below is a possible pipeline:

- Capture training images (you need better resolution than the one displayed above) containing both AMF and DSE.
- Predict colonisation and mycorrhizal structures on training images using the existing CNN1 and CNN2 networks. At each stage, check predictions and convert them to annotations.
- While browsing the images with `amfbrowser`, label microsclerotia as 'hyphopodia' (this is just a name on the interface, `amf` learns from annotated tiles, whatever they contain).
- Re-train CNN2 using `amf train`, the pre-trained CNN2 and your newly-annotated images containing hyphopodia labelling.
- Predict structures on images containing both fungi and check that it works as expected

I would be interested in getting feedback should you decide to try this (I don't have pictures with both fungi to run tests, unfortunately).
